lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/projection/ApplyProjectionUtil.java

```diff
@@ -33,31 +34,38 @@ public final class ApplyProjectionUtil
     private ApplyProjectionUtil() {}
 
     public static List<ConnectorExpression> extractSupportedProjectedColumns(ConnectorExpression expression)
+    {
+        return extractSupportedProjectedColumns(expression, fieldDereference -> true);
+    }
+
+    public static List<ConnectorExpression> extractSupportedProjectedColumns(ConnectorExpression expression, Predicate<FieldDereference> fieldDereferencePredicate)
```

Comment:
Can we make the `fieldDereference` to accept any `ConnectorExpression` instead of just `FieldDereference`